lib/src/model/package_graph.dart

```diff
@@ -869,32 +876,29 @@ class PackageGraph {
   /// set of canonical Libraries).
   Library findButDoNotCreateLibraryFor(Element e) {
     // This is just a cache to avoid creating lots of libraries over and over.
-    if (allLibraries.containsKey(e.library)) {
-      return allLibraries[e.library];
-    }
-    return null;
+    return allLibraries[e.library?.source?.fullName];
   }
 
   /// This is used when we might need a Library object that isn't actually
   /// a documentation entry point (for elements that have no Library within the
   /// set of canonical Libraries).
   Library findOrCreateLibraryFor(DartDocResolvedLibrary resolvedLibrary) {
-    final elementLibrary = resolvedLibrary.library;
-    // This is just a cache to avoid creating lots of libraries over and over.
-    if (allLibraries.containsKey(elementLibrary)) {
-      return allLibraries[elementLibrary];
-    }
+    final libraryElement = resolvedLibrary.library;
     // can be null if e is for dynamic
-    if (elementLibrary == null) {
+    if (libraryElement == null) {
       return null;
     }
-    var foundLibrary = Library.fromLibraryResult(
+    var foundLibrary;
+    foundLibrary = findButDoNotCreateLibraryFor(libraryElement);
```

Comment:
Why declare this on line 891 and then initialize it on 892?

Reply:
No good reason, I just rewrote these sections of code too many times working on this.  Fixed.